src/System.Management.Automation/FormatAndOutput/common/ListWriter.cs

```diff
@@ -36,6 +37,8 @@ internal class ListWriter
         /// <param name="dc">instance of the DisplayCells helper object</param>
         internal void Initialize(string[] propertyNames, int screenColumnWidth, DisplayCells dc)
         {
+            const int propertiesThresHold = OutCommandInner.columnsThresHold;
```

Comment:
I don't think you need this; just use `OutCommandInner.columnsThresHold` instead. C# compiler will use the int value directly in the IL. See an example here:

C# code
---------

```
void Main()
{
	int a = Console.Read();
	string[] b = a <= Test.columnsThresHold ? new string[2] : new string[3];
}

// Define other methods and classes here

internal class Test
{
	internal const int columnsThresHold = 120;
}
```

IL code
--------
```
IL_0000:  nop         
IL_0001:  call        System.Console.Read
IL_0006:  stloc.0     // a
IL_0007:  ldloc.0     // a
IL_0008:  ldc.i4.s    78  // 0x78 == 120
IL_000A:  ble.s       IL_0014
IL_000C:  ldc.i4.3    
IL_000D:  newarr      System.String
IL_0012:  br.s        IL_001A
IL_0014:  ldc.i4.2    
IL_0015:  newarr      System.String
IL_001A:  stloc.1     // b
IL_001B:  ret         

Test..ctor:
IL_0000:  ldarg.0     
IL_0001:  call        System.Object..ctor
IL_0006:  nop         
IL_0007:  ret         
```

Reply:
Thanks for the in-depth explanation!

Fixed.